### ISSUE TITLE
wadouri.loadImageFromPromise(dataSetPromise, ...) fails if the promised dataset is not cached

### ISSUE THREAD

**Issue by adreyfus** [cornerstonejs/cornerstoneWADOImageLoader#184]
`loadImageFromPromise (dataSetPromise, imageId, frame = 0, sharedCacheKey, options, callbacks)` calls `createImage(imageId, pixelData, transferSyntax, options)`, which calls `getImageFrame(imageId)`, which throw `TypeError: Cannot read property 'samplesPerPixel' of undefined` if the dataSet is not cached.

**Comment by swederik** [comment]
Hi Aloïs,

Sorry for the delay. 

I'm not sure I understand how you get to this point in the code if the dataSet is not cached. It appears to me that this createImage call is inside the success handler for the dataSetPromise.

So how could it not be available in the cache at this point? 

https://github.com/adreyfus/cornerstoneWADOImageLoader/blob/129fd4e1957d3f7719ea9b16b39db9714b8d1b6f/src/imageLoader/wadouri/loadImage.js#L25

**Comment by adreyfus** [comment]
Hi @swederik. If the cache is small, the dataset may have been successfully cached, but already purged.

**Comment by dannyrb** [comment]
@adreyfus would you be able to create a minimal reproduction of this? I imagine a sufficiently large Multiframe CT and a small cache size would do the trick?

**Comment by adreyfus** [comment]
> @adreyfus would you be able to create a minimal reproduction of this? I imagine a sufficiently large Multiframe CT and a small cache size would do the trick?

Yes it should do it . I can't try for now because (unfortunately) I'm working on something completely different right now.
